src/Spago/Command/Publish.purs

```diff
@@ -127,7 +128,7 @@ publish _args = do
   let globs = Build.getBuildGlobs { selected: [ selected ], withTests: false, dependencies: allDependencies, depsOnly: false }
   maybeGraph <- Graph.runGraph globs []
   for_ maybeGraph \graph -> do
-    graphCheckErrors <- Graph.toImportErrors selected <$> runSpago (Record.union { graph, selected } env) Graph.checkImports
+    graphCheckErrors <- Graph.toImportErrors selected { reportSrc: true, reportTest: false } <$> runSpago (Record.union { graph, selected } env) Graph.checkImports
```

Comment:
Previously, tests would cause a publish failure here (I think?) if the workspace had pedantic packages.

In this situation, I assume setting `pedantic_packages` to true is desirable because we're publishing. Thoughts? Or should we be using the package's config here?

Reply:
No here it's good to set it to true - this made me spot another bug though: that `for_` there means that if we have no graph then we'd just skip the check. We shouldn't, and throw an error instead 😄

Reply:
It's a two line fix so we can fix it here, unless you'd prefer otherwise

Reply:
What should the error message say if it's not there?

Reply:
I guess we could just print the error we get:
https://github.com/purescript/spago/blob/96c8846042251b26c4668297cde5ed037170c965/src/Spago/Purs.purs#L123-L135